tests/test_edge_cases.py

```diff
@@ -957,15 +957,19 @@ def __init__(self, **data) -> None:
     Foobar(x=1)
 
 
-def test_ignored_type():
-    def foobar():
+def test_type_on_annotation():
+    class FooBar:
         pass
 
     class Model(BaseModel):
-        a: int = foobar
-        b: int
-
-    assert Model.__fields__.keys() == {'b'}
+        a: int = int
+        b: Type[int]
+        c: Type[int] = int
+        d: FooBar = FooBar
+        e: Type[FooBar]
+        f: Type[FooBar] = FooBar
+
+    assert Model.__fields__.keys() == {'b', 'c', 'e', 'f'}
```

Comment:
Any chance you could add a test like what you had in the issue, where you were explicitly confirmed the value of the set property was wrong? Feels like it's worth explicitly checking the behavior is being handled properly

Reply:
Thank you for your review. I add a test case.

Reply:
This looks great! Thanks!